app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchTEIViewModel.kt

```diff
@@ -73,57 +75,68 @@ class SearchTEIViewModel(
-                    displayFrontPageList = searchRepository.getProgram(initialProgramUid)
-                        ?.displayFrontPageList()
-                        ?: false,
-                    canCreateWithoutSearch = searchRepository.canCreateInProgramWithoutSearch(),
-                    queryHasData = queryData.isNotEmpty(),
-                    minAttributesToSearch = searchRepository.getProgram(initialProgramUid)
-                        ?.minAttributesRequiredToSearch()
-                        ?: 0,
-                    isSearching = searching
-                )
-        }
+        _screenState.value = SearchList(
+            previousSate = _screenState.value?.screenState ?: SearchScreenState.NONE,
+            listType = SearchScreenState.LIST,
+            displayFrontPageList = searchRepository.getProgram(initialProgramUid)
+                ?.displayFrontPageList()
+                ?: false,
+            canCreateWithoutSearch = searchRepository.canCreateInProgramWithoutSearch(),
+            isSearching = searching,
+            searchForm = SearchForm(
+                queryHasData = queryData.isNotEmpty(),
+                minAttributesToSearch = searchRepository.getProgram(initialProgramUid)
+                    ?.minAttributesRequiredToSearch()
+                    ?: 1,
+                isForced = shouldOpenSearch,
+                isOpened = shouldOpenSearch
+            ),
+            searchFilters = SearchFilters(
+                hasActiveFilters = _filtersActive.value == true,
+                isOpened = filterIsOpen()
+            )
+        )
     }
 
     fun setMapScreen() {
+        _screenState.value.takeIf { it?.screenState == SearchScreenState.LIST }?.let {
+            searching = (it as SearchList).isSearching
+        }
         _screenState.value = SearchList(
             previousSate = _screenState.value?.screenState ?: SearchScreenState.NONE,
             listType = SearchScreenState.MAP,
             displayFrontPageList = searchRepository.getProgram(initialProgramUid)
                 ?.displayFrontPageList()
                 ?: false,
             canCreateWithoutSearch = searchRepository.canCreateInProgramWithoutSearch(),
-            queryHasData = queryData.isNotEmpty(),
-            minAttributesToSearch = searchRepository.getProgram(initialProgramUid)
-                ?.minAttributesRequiredToSearch()
-                ?: 0,
-            isSearching = searching
+            isSearching = searching,
+            searchForm = SearchForm(
+                queryHasData = queryData.isNotEmpty(),
+                minAttributesToSearch = searchRepository.getProgram(initialProgramUid)
+                    ?.minAttributesRequiredToSearch()
+                    ?: 1,
+                isForced = false,
+                isOpened = false
+            ),
+            searchFilters = SearchFilters(
+                hasActiveFilters = _filtersActive.value == true,
```

Comment:
I think that `_filtersActive.value == true` is used four times in the class, do you think that could be refactored to a method like `private fun hasActiveFilters() = _filtersActive.value == true`?